src/stores/RoomViewStore.tsx

```diff
@@ -353,6 +354,19 @@ export class RoomViewStore extends EventEmitter {
                 });
             }
 
+            // Start a call if requested
+            const currentRoomCall = this.state.roomId ? CallStore.instance.getCall(this.state.roomId) : null;
+            if (payload.view_call && room) {
+                if (!currentRoomCall) {
+                    ElementCall.create(room, false);
+                }
+            }
+            // Destroy the call when leaving call view
+            const prevRoomCall = this.state.roomId ? CallStore.instance.getCall(this.state.roomId) : null;
```

Comment:
`CallStore.instance.getCall(this.state.roomId)` is retuning a different `call` on line 358 & 365 ?